packages/react-app/src/views/TradesListPage.tsx

```
@@ -1,18 +1,82 @@
 
 const TradesListPage: React.FC<TradesListPageProps> = ({ onEditTrade, onLogTransaction }) => {
-  const { trades, isLoadingTrades, errorLoadingTrades, deleteFullTradeInStore } = useAppStore();
+  const { trades, isLoadingTrades, errorLoadingTrades, deleteFullTradeInStore, refreshTrades } = useAppStore();
   const [filterText, setFilterText] = useState('');
+  const [allTransactions, setAllTransactions] = useState<TransactionRecord[]>([]);
+
+  const handleDeleteTransaction = async (transactionId: number) => {
+    if (window.confirm(`Are you sure you want to delete transaction ID ${transactionId}? This action cannot be undone.`)) {
+      try {
+        const result = await window.electronAPI.deleteSingleTransaction(transactionId);
+        if (result && result.success) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)